### PR TITLE
feat: create the GraphQL API of shared data app

### PR DESCRIPTION
**Pull request by cacarrara**
This change creates the GraphQL API for the Shared Data Files app with the
operations to read, update and delete `DataEntry`.

### CODE REVIEW

File: terraso_backend/tests/graphql/mutations/test_shared_data_mutations.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import pytest
+
+from apps.shared_data.models import DataEntry
+
+pytestmark = pytest.mark.django_db
+
+
+def test_data_entry_update_by_creator_works(client_query, data_entries):
+    # Fixtures make sure that the user on client query is the data entry creator
```
Comment (paulschreiber) [inline terraso_backend/tests/graphql/mutations/test_shared_data_mutations.py:9]: suggestion: reword the comment to clarify.

File: terraso_backend/apps/graphql/schema/shared_data.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    id = graphene.ID(source="pk", required=True)
+
+    class Meta:
+        model = DataEntry
+        filter_fields = {
+            "name": ["icontains"],
+            "description": ["icontains"],
+            "slug": ["exact", "icontains"],
+            "url": ["icontains"],
+            "groups__slug": ["icontains"],
```
Comment (paulschreiber) [inline terraso_backend/apps/graphql/schema/shared_data.py:25]: question: should this match `slug`?

File: terraso_backend/apps/graphql/schema/shared_data.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        id = graphene.ID()
+
+    @classmethod
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        user = info.context.user
+        data_entry = DataEntry.objects.get(pk=kwargs["id"])
+
+        if not user.has_perm(DataEntry.get_perm("delete"), obj=data_entry):
+            logger.info(
+                "Attempt to delete a DataEntry, but user has no permission",
```
Comment (paulschreiber) [inline terraso_backend/apps/graphql/schema/shared_data.py:83]: suggestion: reword to `but user lacks permission`

**Review by paulschreiber — APPROVED** [review]

File: terraso_backend/apps/graphql/schema/shared_data.py (hunk 2 above)
Comment (cacarrara) [reply terraso_backend/apps/graphql/schema/shared_data.py:25]: This adds a filter that matches any part of the slug ignoring the word case.

**Comment by cacarrara** [comment]
This is related to #152

**Pull request merged**